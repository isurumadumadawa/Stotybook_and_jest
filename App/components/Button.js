import React from 'react';

import styled from 'styled-components/native';

import colors from '../config/colors';

const ButtonWrapper = styled.TouchableOpacity`
  background-color: ${colors.primary};
  padding-vertical: 14px;
  border-radius: 6px;
  border-width: 1px;
  border-color: ${colors.primary};
  margin-vertical: 7px;
`;

const ButtonWrapperOutline = styled(ButtonWrapper)`
  background-color: transparent;
  border-color: ${colors.border};
`;

const ButtonText = styled.Text`
  color: ${colors.white};
  align-self: center;
  font-size: 18px;
  font-weight: 500;
`;

const ButtonTextOutline = styled(ButtonText)`
  color: ${colors.primary};
`;

export const Button = ({
  onPress = () => {},
  children = '',
  outline = false,
}) => {
  const Button = outline ? ButtonWrapperOutline : ButtonWrapper;
  const Text = outline ? ButtonTextOutline : ButtonText;

  return (
    <Button onPress={onPress}>
      <Text>{children}</Text>
    </Button>
  );
};
