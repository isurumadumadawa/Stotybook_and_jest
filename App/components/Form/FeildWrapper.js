import React from 'react';

import styled from 'styled-components/native';

import colors from '../../config/colors';

const FeildWrapper = styled.View`
  margin-bottom: 20px;
`;

const Label = styled.Text`
  color: ${colors.gray};
  font-size: 18px;
  margin-bottom: 10px;
`;

const MessageText = styled.Text`
  color: ${colors.gray};
  font-size: 13px;
  margin-top: 5px;
`;

const MessageSuccess = styled(MessageText)`
  color: ${colors.success};
`;

const MessageError = styled(MessageText)`
  color: ${colors.error};
`;

export default ({children, label = '', message = '', messageType}) => {
  let Message = MessageText;

  if (messageType == 'success') {
    Message = MessageSuccess;
  } else if (messageType == 'error') {
    Message = MessageError;
  }

  return (
    <FeildWrapper>
      <Label>{label}</Label>
      {children}
      <Message>{message}</Message>
    </FeildWrapper>
  );
};
