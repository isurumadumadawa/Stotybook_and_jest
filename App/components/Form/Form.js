import React from 'react';

import styled from 'styled-components/native';

import colors from '../../config/colors';

const FormWrapper = styled.View`
  margin-horizontal: 20px;
  flex: 1;
`;

const HeaderWrapper = styled.View`
  margin-bottom: 28px;
`;

const Header = styled.Text`
  color: ${colors.primary};
  font-weight: 600;
  font-size: 32px;
  margin-bottom: 12px;
`;

const SubHeader = styled.Text`
  color: ${colors.gray};
  font-size: 20px;
  margin-bottom: 12px;
`;

export default ({children, header, subheader}) => (
  <FormWrapper>
    {(header || subheader) && (
      <HeaderWrapper>
        {header && <Header>{header}</Header>}
        {subheader && <SubHeader>{subheader}</SubHeader>}
      </HeaderWrapper>
    )}

    {children}
  </FormWrapper>
);
