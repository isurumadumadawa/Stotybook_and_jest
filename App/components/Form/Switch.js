import React from 'react';
import {Switch} from 'react-native';

import FeildWrapper from './FeildWrapper';

export default ({label, message, messageType, ...otherProps}) => (
  <FeildWrapper label={label} message={message} messageType={messageType}>
    <Switch {...otherProps} />
  </FeildWrapper>
);
