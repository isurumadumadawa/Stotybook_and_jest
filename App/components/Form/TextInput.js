import React from 'react';

import styled from 'styled-components/native';

import colors from '../../config/colors';

import FeildWrapper from './FeildWrapper';

const Input = styled.TextInput`
  font-size: 14px;
  font-weight: 500;
  padding-bottom: 10px;
`;

const Boarder = styled.View`
  height: 1px;
  background-color: ${colors.border};
`;

export default ({label, message, messageType, ...otherProps}) => (
  <FeildWrapper label={label} message={message} messageType={messageType}>
    <Input {...otherProps} />
    <Boarder />
  </FeildWrapper>
);
