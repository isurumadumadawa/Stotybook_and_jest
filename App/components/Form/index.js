import Form from './Form';
import FeildWrapper from './FeildWrapper';
import TextInput from './TextInput';
import Switch from './Switch';

export {Form, FeildWrapper, TextInput, Switch};
