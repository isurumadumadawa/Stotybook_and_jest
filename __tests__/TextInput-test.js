import 'react-native';
import React from 'react';

import render from 'react-test-renderer';
import {expect} from '@jest/globals';

import {TextInput} from '../App/components/Form';

test('Button snapshot', () => {
  const snapShot = render.create(<TextInput />).toJSON();
  expect(snapShot).toMatchSnapshot();
});
