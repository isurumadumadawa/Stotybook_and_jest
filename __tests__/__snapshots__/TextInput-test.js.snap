// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button snapshot 1`] = `
<View
  style={
    Array [
      Object {
        "marginBottom": 20,
      },
    ]
  }
>
  <Text
    style={
      Array [
        Object {
          "color": "#9ca5ab",
          "fontSize": 18,
          "marginBottom": 10,
        },
      ]
    }
  >
    
  </Text>
  <TextInput
    allowFontScaling={true}
    rejectResponderTermination={true}
    style={
      Array [
        Object {
          "fontSize": 14,
          "fontWeight": "500",
          "paddingBottom": 10,
        },
      ]
    }
    underlineColorAndroid="transparent"
  />
  <View
    style={
      Array [
        Object {
          "backgroundColor": "#c6c6c6",
          "height": 1,
        },
      ]
    }
  />
  <Text
    style={
      Array [
        Object {
          "color": "#9ca5ab",
          "fontSize": 13,
          "marginTop": 5,
        },
      ]
    }
  >
    
  </Text>
</View>
`;
